### ISSUE TITLE
Libero Spatial eval reproducibility

### ISSUE THREAD

**Issue by KJW988** [Zhangwenyao1/DreamVLA#16]
Hello, thank you for sharing your work and for releasing the code and checkpoints.

I understand that some small performance differences can arise from evaluation variance, rollout initialization randomness, and PyTorch/cuDNN non-determinism. However, I wanted to ask about the reproducibility of the LIBERO Spatial evaluation.

I evaluated the released checkpoint and found that most per-task results look reasonably consistent, but task 5 is noticeably lower than the others:
	•	task 5: pick_up_the_black_bowl_on_the_ramekin_and_place_it_on_the_plate
	•	success rate from my run: 20.0%

Since most of the other tasks are much higher, I was wondering whether this task 5 result is still within the normal range of evaluation variance, or whether it might indicate that something is inconsistent in my evaluation setup.

I also wanted to confirm whether the random seed is supposed to be fixed to 66 for all LIBERO evaluations.

For reference, here are my per-task results for LIBERO Spatial:
	•	task 0: 100.0%
	•	task 1: 100.0%
	•	task 2: 95.0%
	•	task 3: 100.0%
	•	task 4: 95.0%
	•	task 5: 20.0%
	•	task 6: 100.0%
	•	task 7: 85.0%
	•	task 8: 100.0%
	•	task 9: 90.0%
	•	Avg.: 88.5%

For the other task suites, I reproduced the following results:
	•	Object: 89.5
	•	Goal: 90.0
	•	Long: 86.5

So my main questions are:
	1.	Is fixing the random seed to 66 the correct official setting for LIBERO evaluation?
	2.	Is a result around 20.0% on Spatial task 5 still considered normal for the released checkpoint?

I would appreciate any clarification. Thank you.

**Comment by KJW988** [comment]
I used the following evaluation script.
I only made minor modifications to the path settings and related configurations.

```
#!/bin/bash

source /opt/conda/etc/profile.d/conda.sh
conda activate dreamvla
export WANDB_MODE=offline
export MUJOCO_GL=egl
export PYOPENGL_PLATFORM=egl

resume_from_checkpoint="/workspace/DreamVLA/checkpoints/libero_spatial_97.5.pth"
vit_checkpoint_path="/workspace/DreamVLA/checkpoints/vit_mae/mae_pretrain_vit_base.pth"
save_checkpoint_path="/workspace/DreamVLA/eval_outputs"

test_id=$(basename "$resume_from_checkpoint" .pth)
LOG_DIR="${save_checkpoint_path}/${test_id}"
mkdir -p "${LOG_DIR}"
logfile="${LOG_DIR}/eval.log"

node=1
node_num=4

python -m torch.distributed.run  --nnodes=${node} --nproc_per_node=${node_num} --master_port=10133 eval_libero.py \
    --traj_cons \
    --rgb_pad 10 \
    --gripper_pad 4 \
    --gradient_accumulation_steps 1 \
    --bf16_module "vision_encoder" \
    --vit_checkpoint_path ${vit_checkpoint_path} \
    --calvin_dataset "" \
    --libero_path "/workspace/DreamVLA/LIBERO" \
    --workers 16 \
    --lr_scheduler cosine \
    --save_every_iter 50000 \
    --num_epochs 20 \
    --seed 66 \
    --batch_size 64 \
    --precision fp32 \
    --weight_decay 1e-4 \
    --num_resampler_query 16 \
    --run_name test \
    --transformer_layers 24 \
    --hidden_dim 1024 \
    --transformer_heads 16 \
    --phase "evaluate" \
    --finetune_type libero_spatial \
    --action_pred_steps 3 \
    --future_steps 3 \
    --sequence_length 7 \
    --obs_pred \
    --gripper_width \
    --eval_libero_ensembling \
    --use_dit_head \
    --load_track_labels \
    --load_sam_features \
    --sam_feat_pred \
    --loss_sam_feat \
    --flow_as_mask \
    --attn_implementation "sdpa" \
    --save_checkpoint_path ${save_checkpoint_path} \
    --resume_from_checkpoint ${resume_from_checkpoint} | tee ${logfile}

```

**Comment by Zhangwenyao1** [comment]
Hello,

Thank you for the careful evaluation and for sharing the detailed results.

Yes, for our LIBERO evaluation we fix the random seed to 66, so that is the correct setting to use.

Regarding Spatial task 5 (`pick_up_the_black_bowl_on_the_ramekin_and_place_it_on_the_plate`), LIBERO can be somewhat unstable, and this task in particular may show larger variance than the others. Because of that, a low result on this single task does not necessarily mean that your evaluation setup is incorrect.

My guess is that the number of evaluation trials may simply be too small to get a stable estimate for this task. I would suggest running the evaluation a few more times and checking whether the result becomes more consistent across runs.

Since your other Spatial tasks and the other suites are generally reproduced well, your overall setup looks reasonably consistent to me.

More broadly, if reproducibility is a main concern, I would also recommend evaluating on CALVIN, which is generally more reproducible and tends to be more stable across runs than LIBERO.

Thank you again for checking this so carefully.

Best,
Wenyao

**Comment by KJW988** [comment]
Hello,

Thank you very much for your detailed and helpful reply.

The results I mentioned previously were obtained using seed 66 with 20 evaluation episodes.

Following your suggestion, I increased the number of evaluation episodes to 50 and re-ran the evaluation with the same seed (66). The updated results are as follows:

- Spatial: 89.4
- Object: 92.2
- Goal: 90.4
- Long: 86.2

In addition, for task 5 in the Spatial task suite, which was the main issue I raised before, I again observed a similarly low success rate of 24%. This makes me think that, as you mentioned, the result is likely due to the characteristics and variance of that particular task rather than a problem in my evaluation setup.

For future reproducibility checks, I will also evaluate on CALVIN as you recommended.

Thank you again for your kind guidance.

Best,
Jiwon
